sklearn/manifold/mds.py

```diff
@@ -129,18 +131,18 @@ def _smacof_single(similarities, metric=True, n_components=2, init=None,
     return X, stress, it + 1
 
 
-def smacof(similarities, metric=True, n_components=2, init=None, n_init=8,
+def smacof(dissimilarities, metric=True, n_components=2, init=None, n_init=8,
```

Comment:
I don't expect many people to pass positional args with the kwarg notation but maybe we could change the prototype of the function to support backward compat to be on the safe side:

```python
def smacof(dissimilarities=None, similarities=None, metric=True, ...):
    if similarities is not None and dissimilarities is None:
        # issue DeprecationWarning and then set:
        dissimilarities = similarities
    if dissimilarities is None and similarities is None:
        raise ValueError("dissimilarities is a mandatory argument")
```

Please `git grep DeprecationWarning` to find examples of informative deprecation warnings in the code base.